src/BenchmarkDotNet/Validators/ExecutionValidatorBase.cs

```diff
@@ -24,29 +25,34 @@ public IEnumerable<ValidationError> Validate(ValidationParameters validationPara
 
             foreach (var typeGroup in validationParameters.Benchmarks.GroupBy(benchmark => benchmark.Descriptor.Type))
             {
-                if (!TryCreateBenchmarkTypeInstance(typeGroup.Key, errors, out var benchmarkTypeInstance))
-                {
-                    continue;
-                }
+                var executors = new List<BenchmarkExecutor>();
 
-                if (!TryToSetParamsFields(benchmarkTypeInstance, errors))
+                foreach (var benchmark in typeGroup)
                 {
-                    continue;
-                }
+                    if (!TryCreateBenchmarkTypeInstance(typeGroup.Key, errors, out var benchmarkTypeInstance))
+                        continue;
 
-                if (!TryToSetParamsProperties(benchmarkTypeInstance, errors))
-                {
-                    continue;
-                }
+                    if (!TryToFillParameters(benchmark, benchmarkTypeInstance, errors))
+                        continue;
 
-                if (!TryToCallGlobalSetup(benchmarkTypeInstance, errors))
-                {
-                    continue;
+                    if (!TryToCallGlobalSetup(benchmarkTypeInstance, errors))
+                        continue;
+
+                    if (!TryToCallIterationSetup(benchmarkTypeInstance, errors))
+                        continue;
+
+                    executors.Add(new BenchmarkExecutor(benchmarkTypeInstance, benchmark));
```

Comment:
You added an internal foreach loop, so these `continue;` aren't continuing on the outer foreach loop that they were before. I wonder if that's contributing to the deadlock you are seeing.